addons/survey/static/src/js/survey_session_manage.js

```diff
@@ -318,14 +328,29 @@ publicWidget.registry.SurveySessionManage = publicWidget.Widget.extend({
             delete this.resultsRefreshInterval;
         }
 
-        Promise.all([fadeOutPromise, nextQuestionPromise]).then(function (results) {
+        Promise.all([fadeOutPromise, nextQuestionPromise]).then(async function (results) {
             if (results[1]) {
                 var $renderedTemplate = $(results[1]);
                 self.$el.replaceWith($renderedTemplate);
-                self.attachTo($renderedTemplate);
-                self.$el.fadeIn(self.fadeInOutTime, function () {
+
+                // Ensure new question is fully loaded before force loading last question screen.
+                await self.attachTo($renderedTemplate);
+                if (goBack) {
+                    // As we arrive on first "question" screen, simulate going to the results or leaderboard.
+                    self._setShowInputs(true);
+                    self._setShowAnswers(true);
+                    self.currentScreen = 'results';
+                    var nextScreen = self._getNextScreen();
+                    if (nextScreen === 'leaderboard') {
```

Comment:
Setting the next screen to result, just to check next screen, and then change it again.
Is there not a more straight forward way to do this?
Like checking the props the the question directly?

This is a bit "tiré par les cheveux".